lib/cloudflare_protected_email_address.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'cgi'
+
+class CloudflareProtectedEmailAddress
+  def initialize(obfuscated)
+    @obfuscated = obfuscated
+  end
+
+  def human_readable
+    return CGI.unescape(escaped_email_address) if CGI.unescape(escaped_email_address).include?('@')
+    obfuscated
+  end
+
+  private
+
+  attr_accessor :obfuscated
+
+  def obfuscated_characters
+    obfuscated.scan(/.{2}/).drop(1).map(&:hex)
+  end
+
+  def key
+    obfuscated[0..1].hex
+  end
+
+  def escaped_email_address
+    obfuscated_characters.reduce('') do |email, encoded_character|
+      email + (encoded_character ^ key).to_s(16).prepend('%')
+    end
```

Comment:
This is a bit over-complicated. 

Perhaps a simple map+join would be simpler/clearer?

```ruby
   obfuscated_characters.map { |char| (char ^ key).to_s(16).prepend('%') }.join
```